File: app/src/main/java/com/example/sergiobelda/photoeditor/editableimageview/StrategyTool.java
package com.example.sergiobelda.photoeditor.editableimageview;

import android.view.MotionEvent;

abstract class StrategyTool {
    EditableImageView imageView;

    void setImageView(EditableImageView imageView) {
        this.imageView = imageView;
    }

    void invalidate() {
        imageView.invalidate();
    }

    abstract void onTouchEvent(MotionEvent event);
}

File: app/src/main/java/com/example/sergiobelda/photoeditor/editableimageview/StrategyTool.kt
package com.example.sergiobelda.photoeditor.editableimageview

import android.view.MotionEvent

abstract class StrategyTool {
    abstract var editableImageView: EditableImageView

    fun setImageView(editableImageView: EditableImageView) {
        this.editableImageView = editableImageView;
    }

    fun invalidate() {
        editableImageView.invalidate()
    }

    abstract fun onTouchEvent(event: MotionEvent)
}